src/pages/Dashboard.tsx

```diff
@@ -1,35 +1,132 @@
+
+    useEffect(() => {
+        (async () => {
+            try {
+                const storedRequests = await getFiledRequests(TEST_ORGANIZATION, TEST_GROUP);
+                setDisplayedRequests({...displayedRequests, ...storedRequests});
+            } catch (error: unknown) {
+                console.log("Fetching requests broken very sad")
+            }
+            try {
+                const storedInsights = await getInsights(TEST_ORGANIZATION, TEST_GROUP);
+                setDisplayedInsights({...displayedInsights, ...storedInsights});
+            } catch (error: unknown) {
+                console.log("Fetching insights broken very sad")
+            }
+        })();
+    }, [])
+
+    // Generate and store hardcoded insight object for now
+    const createNewInsight = async () => {
+        const hardcodedInsight: Insight = {
+            createdBy: 'userId',
+            createDate: 1,
+            lastUpdated: 2,
+            ticketId: 'DATASCI-2864:',
+            title: 'FireBase - Title blah blah blah',
+            imgs: [{src: Logo, alt: 'google image'}],
+            outcomeNumber: 21,
+            keyStat: '21% increase per annum',
+            keyNumber: '21%',
+            tldr: 'string string string string string string string string string',
+            takeaway: 'string string string string string string string string',
+            requests: ['RequestId1'],
+            collaborators: ['UserId1']
+        };
+        try {
+            const insightId: string = await addInsight(TEST_ORGANIZATION, TEST_GROUP, hardcodedInsight);
+            const newInsightEntry: {[id: string]: Insight} = {[insightId]: hardcodedInsight}
+            setDisplayedInsights({...displayedInsights, ...newInsightEntry});
+        } catch (error: unknown) {
+            // Replace with user visible messaging
+            console.log("Could not create new insight");
+        }
+    }
+
+    // Generate and store hardcoded request object for now
+    const createNewFiledRequest = async () => {
+        const hardcodedFiledRequest: FiledRequest = {
+            createdBy: "BigBossId",
+            createDate: 123,
+            lastUpdated: 123,
+            title: "FireBase - Title blah blah blah",
+            assignee: "CodeMonkeyId",
+            priority: "High",
+            status: "Assigned",
+            deadline: 123
+        };
+        try {
+            const filedRequestId: string = await addFiledRequest(TEST_ORGANIZATION, TEST_GROUP, hardcodedFiledRequest);
+            const newRequestEntry: {[id: string]: FiledRequest} = {[filedRequestId]: hardcodedFiledRequest}
+            setDisplayedRequests({...displayedRequests, ...newRequestEntry});
+        } catch (error: unknown) {
+            // Replace with user visible messaging
+            console.log("Could not create new request");
+        }
+    }
+
+    const generateDeleteFiledRequestHandler = (filedRequestId: string): (() => void) => {
```

Comment:
Sorry I'm still a bit confused, what does a generator give us that the following doesn't
```
consst deleteFiledRequestHandler = async(filedRequestId: string) => {
try{...}
catch{...}
}```

Reply:
In the current implementation, which could be changed, the filedRequest/Insight id isn't passed to the FiledRequest/Insight card, so the delete handlers would need to be initialized for each id before being passed to the cards. The generator is then used to create these delete handlers in TicketFeedContainer when iterating through FiledRequests/Insights